qmail-smtpd.c

```diff
@@ -230,7 +233,12 @@ void smtp_helo(arg) char *arg;
 }
 void smtp_ehlo(arg) char *arg;
 {
-  smtp_greet("250-"); out("\r\n250-PIPELINING\r\n250 8BITMIME\r\n");
+  unsigned int i;
+  smtp_greet("250-");
```

Comment:
what about `out("\r\n");` here? So that the ehlo_gen[] functions do not have to begin their output by printing unfinished work of previous ehlo_gen?

There will be `\r\n` printed at the end anyway.

Reply:
You are absolutely right. This was a microoptimization from DJB that avoids one function call. I moved it into ```smtp_greet()```, where it makes more sense.